content/6_Advanced/Catalan.mdx

```diff
@@ -0,0 +1,164 @@
+---
+id: catalan
+title: 'Catalan Numbers'
+author: Mihnea Brebenel
+description: ''
+prerequisites:
+  - combo
+frequency: 1
+---
+
+<Resources>
+	<Resource
+		source="cp-algo"
+		title="Catalan Numbers"
+		url="https://cp-algorithms.com/combinatorics/catalan-numbers.html"
+		starred
+	>
+		Well documented article.
+	</Resource>
+</Resources>
+
+The [Catalan numbers](https://en.wikipedia.org/wiki/Catalan_number) are a sequence of positive integers that can be very useful in counting problems in combinatorics.
+The $n$-th Catalan can be expressed as following using binomial coeffiecients:
+
+$$
+C_n=\frac{1}{n+1}\cdot \binom{2n}{n}=\frac{(2n)!}{(n+1)!\,n!}
+$$
+
+They also have the recurrence formula
+$$
+C_{n+1}= \sum^{n}_{i=0}{C_i \cdot C_{n-i}} \,\,\, \text{ for } n \ge 0 \\
+$$
+which can also be expressed as
+$$
+C_n=\frac{2(2n-1)}{n+1} \cdot C_{n-1}
+$$
+
+The first $5$ Catalan numbers are
+
+<center>
+
+| n | 0| 1 | 2 | 3 | 4 | 5 |
+|---| -| - | - | - | - | - |
+| $C_n$| 1 | 1 | 2 | 5 | 14| 42 |
+
+</center>
+
+# Applications
+
+The Catalan numbers can be used to represent a wide variety of things.
+
+For example, $C_n$ is equal to the number of valid parenthesis expressions of length $2n$.
+Take, for instance, $C_3=5$:
+* `()()()`
+* `(())()`
+* `()(())`
+* `((()))`
+* `(()())`
+
+It's also equal to the number of full binary trees with $n+1$ leaves.
+The following image shows the $5$ binary trees with $4$ leaves:
+
+![binary-trees](./assets/catalan-binary-trees.png)
+
+$C_n$ is also the number of monotonic lattice paths along the edges of a $n \times n$ grid that don't pass above the diagonal.
+The paths start in the lower left corner and ends in the upper right corner.
+
+For example, there are $C_4=14$ paths in a $4 \times 4 grid$:
+
+![lattice points](./assets/catalan-lattice-paths.png)
```

Comment:
i think there was a github issue/comment about this, but the bottom left image is wrong

could you fix it? or i can do some image editing lol

Reply:
Solved
<img width="139" alt="image" src="https://github.com/user-attachments/assets/45285f82-5a03-461d-8834-9f99706f26e7">